xmtp_mls/src/storage/sql_key_store.rs

```diff
@@ -1,81 +1,1008 @@
+                Err(e)
+            }
         }
+    }
+
+    fn delete_encryption_epoch_key_pairs<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        EpochKey: traits::EpochKey<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        epoch: &EpochKey,
+        leaf_index: u32,
+    ) -> Result<(), Self::Error> {
+        let key = epoch_key_pairs_id(group_id, epoch, leaf_index)?;
+        self.delete::<CURRENT_VERSION>(EPOCH_KEY_PAIRS_LABEL, &key)
+    }
+
+    fn clear_proposal_queue<GroupId: traits::GroupId<CURRENT_VERSION>>(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<(), Self::Error> {
+        let key = build_key::<CURRENT_VERSION, &GroupId>(QUEUED_PROPOSAL_LABEL, group_id);
+        self.delete::<CURRENT_VERSION>(QUEUED_PROPOSAL_LABEL, &key);
+
         Ok(())
     }
+
+    fn mls_group_join_config<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        MlsGroupJoinConfig: traits::MlsGroupJoinConfig<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<Option<MlsGroupJoinConfig>, Self::Error> {
+        self.read(JOIN_CONFIG_LABEL, &serde_json::to_vec(group_id).unwrap())
+    }
+
+    fn write_mls_join_config<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        MlsGroupJoinConfig: traits::MlsGroupJoinConfig<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        config: &MlsGroupJoinConfig,
+    ) -> Result<(), Self::Error> {
+        let key = serde_json::to_vec(group_id).unwrap();
+        let value = serde_json::to_vec(config).unwrap();
+
+        self.write::<CURRENT_VERSION>(JOIN_CONFIG_LABEL, &key, &value)
+    }
+
+    fn own_leaf_nodes<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        LeafNode: traits::LeafNode<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<Vec<LeafNode>, Self::Error> {
+        self.read_list(OWN_LEAF_NODES_LABEL, &serde_json::to_vec(group_id).unwrap())
+    }
+
+    fn append_own_leaf_node<
+        GroupId: traits::GroupId<CURRENT_VERSION>,
+        LeafNode: traits::LeafNode<CURRENT_VERSION>,
+    >(
+        &self,
+        group_id: &GroupId,
+        leaf_node: &LeafNode,
+    ) -> Result<(), Self::Error> {
+        let key = serde_json::to_vec(group_id)?;
+        let value = serde_json::to_vec(leaf_node)?;
+        self.append::<CURRENT_VERSION>(OWN_LEAF_NODES_LABEL, &key, &value)
+    }
+
+    fn clear_own_leaf_nodes<GroupId: traits::GroupId<CURRENT_VERSION>>(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<(), Self::Error> {
+        let key = serde_json::to_vec(group_id)?;
+        self.delete::<CURRENT_VERSION>(OWN_LEAF_NODES_LABEL, &key)
+    }
+
+    fn aad<GroupId: traits::GroupId<CURRENT_VERSION>>(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<Vec<u8>, Self::Error> {
+        let key = serde_json::to_vec(group_id)?;
+        self.read_list(AAD_LABEL, &key)
```

Comment:
I know this is how it's done in the MemoryStorage, but it's a little weird to me that we are using the single `write` for aad but `read_list`.

Is there meant to be one `aad` or many per key? Maybe it actually all works because everything is a JSON array under the hood.

Reply:
I agree something doesn't seem to checkout here. 🤔 I think it should just be read not read_list.. @franziskuskiefer do you have thoughts here?

Reply:
Yeah, this looks like it should really be a `read`. Good catch. I'll check the memory store in openmls.